### PR TITLE
Datagen: Special-case cfg(test) in registry macro to fix misleading warnings

### PR DESCRIPTION
**Pull request by sffc**
Fixes https://github.com/unicode-org/icu4x/issues/4296

### CODE REVIEW

File: provider/datagen/src/registry.rs [hunk 1]
```diff
@@ -491,6 +504,21 @@ registry!(
...
+///
+/// ✨ *Enabled with the `legacy_api` Cargo feature.*
+#[deprecated(
+    since = "1.3.0",
+    note = "use `all_keys` with the required cargo features"
+)]
+#[cfg(feature = "legacy_api")]
+pub fn all_keys_with_experimental() -> Vec<DataKey> {
+    all_keys()
+}
```
Comment (sffc) [inline provider/datagen/src/registry.rs:520]: I pulled this out of the macro because it didn't need to be in the macro

**Comment by sffc** [comment]
Autosubmit = yes

**Review by Manishearth — APPROVED** [review]
clever

**Comment by robertbastian** [comment]
lgtm

**Pull request merged**